### PR TITLE
Contact groups and channel should be references instead of single UUIDs

### PR DESCRIPTION
**Pull request by rowanseymour**
Like how they are elsewhere

### CODE REVIEW

File: flows/contact.go [hunk 1]
```diff
@@ -196,14 +196,14 @@ type fieldValueEnvelope struct {
...
-	Fields      map[FieldKey]fieldValueEnvelope `json:"fields,omitempty"`
-	ChannelUUID ChannelUUID                     `json:"channel_uuid,omitempty" validate:"omitempty,uuid4"`
+	UUID     ContactUUID                     `json:"uuid" validate:"required,uuid4"`
+	Name     string                          `json:"name"`
+	Language utils.Language                  `json:"language"`
+	Timezone string                          `json:"timezone"`
+	URNs     URNList                         `json:"urns"`
+	Groups   []*GroupReference               `json:"groups,omitempty" validate:"dive"`
+	Fields   map[FieldKey]fieldValueEnvelope `json:"fields,omitempty"`
+	Channel  *ChannelReference               `json:"channel,omitempty" validate:"omitempty,dive"`
```
Comment (rowanseymour) [inline flows/contact.go]: possibly these shouldn't be pointers, but if not that's for another day as they are pointers everywhere else

**Comment by nicpottier** [comment]
What's the argument again for having references over UUIDs?

**Comment by rowanseymour** [comment]
@nicpottier we use references elsewhere like on input/events/actions. Events and actions did use to be different in that regard but you pushed for making them consistent.

**Comment by nicpottier** [comment]
But what was our reasoning again? Why do we want refs as opposed to UUIDs? Maybe we should document these decisions and the thought behind them.

**Comment by rowanseymour** [comment]
@nicpottier consistency and readability was what I recall

**Comment by rowanseymour** [comment]
@nicpottier if we're formulating a rule - I don't think every like between two things in the session session needs to be a reference object, e.g. `run.parent_uuid`. But it is kinda nice for references to assets which aren't defined in the session JSON.

**Comment by nicpottier** [comment]
So what's the hard rule as to when a naked UUID is ok vs not ok?

**Comment by rowanseymour** [comment]
@nicpottier use a reference object when referencing an asset (e.g. group, label). Use a naked UUID when referencing something defined in the session itself (e.g. run, step)

**Comment by nicpottier** [comment]
k, that works.

**Pull request merged**